src/renderer/editor/clipboard.ts

```diff
@@ -46,7 +46,12 @@ export const pasteText: () => void = () => {
   const { clipboard } = store.getState();
 
   // Paste after the last word in the selection
-  const { endIndex } = range;
+  const { startIndex, endIndex } = range;
+
+  if (clipboard.length && endIndex - startIndex > 1) {
+    // only replace if you're highlighting more than one word.
+    dispatchOp(makeDeleteSelection(range));
```

Comment:
The problem with dispatching two ops at once (this one plus the paste word one) is that it requires two separate undos. Instead, make a new op type that does both the delete selection and then the paste word action in a single op.

Reply:
I reimplemented it at the reducer level, since it was kind of annoying to do at the op level, check [commit 91f2edb](https://github.com/chloebrett/mlvet/pull/286/commits/91f2edb91a89ac186f480b3d88431408eb1d3c4e) if you care enough.

At the reducer level, I save having to specify the precise differences between a paste which deletes, and a paste which doesn't.